### PR TITLE
SearchCommand: Improve on search functionality

### LINKED ISSUE #151: Add Find to UML diagram, Java class and test file

### LINKED ISSUE #166: Rename Find command to Search Command

### PR DESCRIPTION
**Pull request by khsc96**
User can now search keyword base on any fields present except for Transaction id and cash on delivery field.
User can also specify keyword to only specifically search for that particular field by providing corresponding prefix.

### CODE REVIEW

**Comment by Cherweijie** [comment]
As discussed, we will review this again.

**Review by Cherweijie — APPROVED** [review]
As discussed, we will review this again later.

**Review by Cherweijie — COMMENTED** [review]
As discussed, we will review this again.

File: src/main/java/seedu/address/logic/parser/SearchCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            keywordsString = addPrefixKeywordsToList(keywordsString, argMultimap, PREFIX_COMMENT);
+        }
+
+        if (keywordContainsOrderPrefix.getHasCod()) {
+            keywordsString = addPrefixKeywordsToList(keywordsString, argMultimap, PREFIX_COD);
+        }
+
+        if (keywordContainsOrderPrefix.getHasItemType()) {
+            keywordsString = addPrefixKeywordsToList(keywordsString, argMultimap, PREFIX_TYPE);
+        }
```
Comment (JeremyLoh) [inline src/main/java/seedu/address/logic/parser/SearchCommandParser.java]: I think can extract this method out into another function, hopefully will solve the codacy error of NPath complexity 
`The method 'parse(String)' has an NPath complexity of 384`.

**Review by JeremyLoh — APPROVED** [review]
LGTM! Good job. :+1:

**Comment by khsc96** [comment]
Fixed merge conflict please help review thankyou! @JeremyLoh @Cherweijie

**Review by JeremyLoh — APPROVED** [review]
LGTM. Thanks!

**Comment by Cherweijie** [comment]
LGTM. Good job, Kenny!

**Review by Cherweijie — APPROVED** [review]
LGTM. Good job, Kenny!

**Pull request merged**